language/tools/move-mv-llvm-compiler/tests/rbpf-tests.rs

```diff
@@ -177,10 +260,120 @@ fn run_rbpf(exe: &Path) -> anyhow::Result<()> {
             // currently emit a main function that returns void, so this value
             // is seemingly whatever happens to be in the return register.
         }
+        Err(EbpfError::UserError(e)) if e.is::<rbpf_setup::AbortError>() => {
+            if let Some(expected_code) = test_plan.abort_code() {
+                let last_event = context_object.events.last();
+                match last_event {
+                    Some(rbpf_setup::Event::LogU64(c1, c2, c3, c4, c5)) => {
+                        assert!(
+                            [c1, c2, c3, c4, c5].iter().all(|c| *c == c1),
+                            "all abort codes same"
+                        );
+                        if *c1 != expected_code {
+                            panic!("unexpected abort code {c1}, expected {expected_code}");
+                        }
+                    }
+                    _ => {
+                        panic!("abort without abort code?!");
+                    }
+                }
+            } else {
+                panic!("test aborted unexpectedly");
+            }
+        }
         e => {
             panic!("{e:?}");
         }
     }
 
     Ok(())
 }
+
+mod rbpf_setup {
+    use super::rbpf;
+    use anyhow::anyhow;
+    use rbpf::error::EbpfError;
+    use rbpf::memory_region::MemoryMapping;
+    use rbpf::vm::*;
+    use std::sync::Arc;
+
+    #[derive(Default, Debug)]
+    pub struct Context {
+        pub events: Vec<Event>,
+    }
+
+    impl ContextObject for Context {
+        fn trace(&mut self, _state: [u64; 12]) {}
+        fn consume(&mut self, _amount: u64) {}
+        fn get_remaining(&self) -> u64 {
+            u64::max_value()
+        }
+    }
+
+    #[derive(Debug)]
+    pub enum Event {
+        LogU64(u64, u64, u64, u64, u64),
+    }
+
+    pub fn build_loader() -> anyhow::Result<Arc<BuiltInProgram<Context>>> {
+        let config = Config {
+            dynamic_stack_frames: false,
+            enable_elf_vaddr: false,
+            reject_rodata_stack_overlap: false,
+            static_syscalls: false,
+            enable_instruction_meter: false,
+            ..Config::default()
+        };
+        let mut loader = BuiltInProgram::new_loader(config);
+
+        loader
+            .register_function_by_name("abort", SyscallAbort::call)
+            .map_err(|e| anyhow!("{e}"))?;
+        loader
+            .register_function_by_name("sol_log_64_", SyscallLogU64::call)
+            .map_err(|e| anyhow!("{e}"))?;
+
+        Ok(Arc::new(loader))
+    }
+
+    pub struct SyscallAbort;
+
+    impl SyscallAbort {
```

Comment:
do we need some documentation on the specification of abort?

Reply:
It could be nice. I'll have to think about where it would be appropriate to document. These are both Solana syscalls, and they are essentially undocumented within Solana itself.